frontend/src/components/common/header/header.tsx

```diff
@@ -3,26 +3,40 @@ import logo from 'assets/img/logo.svg';
 import { AppRoute } from 'common/enums/enums';
 import { FC } from 'common/types/types';
 import { Button, Image } from 'components/common/common';
-import { useAppSelector, useState } from 'hooks/hooks';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useLocation,
+  useState,
+} from 'hooks/hooks';
+import { dashboardActions } from 'store/actions';
 
-import { Popup } from './components/components';
+import { Popup, SearchBar } from './components/components';
 import styles from './styles.module.scss';
 
 const Header: FC = () => {
   const [isMenuPopupVisible, setIsMenuPopupVisible] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
   const { user } = useAppSelector((state) => state.auth);
+  const location = useLocation();
 
+  const isRoot = location.pathname === AppRoute.ROOT;
   const hasUser = Boolean(user);
 
   const handlePopupOpen = (): void =>
     setIsMenuPopupVisible(!isMenuPopupVisible);
 
+  const onSearch = (search: string): void => {
```

Comment:
```suggestion
  const handleSearch = (search: string): void => {
```
`handle` for local callback’s